### PR TITLE
JointAxis: update calls to use sdf::Errors output

### PR DESCRIPTION
**Pull request by marcoag**
Signed-off-by: Marco A. Gutierrez <marco@openrobotics.org>

# 🎉 New feature

Work towards https://github.com/gazebosim/sdformat/issues/820.

Depends on: https://github.com/gazebosim/sdformat/pull/1141.

## Summary
Adds missing `Errors` structure parameters in a few methods of the JointAxis class.

## Test it

Using the JointAxis class should report all errors through `sdf::Errors` if the `errors` parameter is used, it should print them otherwise.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: include/sdf/JointAxis.hh [hunk 1]
```diff
@@ -229,6 +229,16 @@ namespace sdf
...
     public: sdf::ElementPtr ToElement(unsigned int _index = 0u) const;
 
+    /// \brief Create and return an SDF element filled with data from this
+    /// joint axis.
+    /// Note that parameter passing functionality is not captured with this
+    /// function.
+    /// \param[out] _errors Vector of errors.
+    /// \param[in] _index Index of this joint axis
+    /// \return SDF element pointer with updated joint values.
+    public: sdf::ElementPtr ToElement(sdf::Errors &_errors,
```
Comment (aaronchongth) [inline include/sdf/JointAxis.hh:239]: I see `JointAxis_TEST` does not have a `ToElement` test so far, could you add some together with this overload?

File: src/JointAxis.cc [hunk 2]
```diff
@@ -374,37 +375,54 @@ sdf::ElementPtr JointAxis::Element() const
...
-  sdf::ElementPtr xyzElem = axisElem->GetElement("xyz");
-  xyzElem->Set<gz::math::Vector3d>(this->Xyz());
+  sdf::ElementPtr axisElem = elem->GetElement(axisElemName, _errors);
+  sdf::ElementPtr xyzElem = axisElem->GetElement("xyz", _errors);
+  xyzElem->Set<gz::math::Vector3d>(_errors, this->Xyz());
   if (!this->XyzExpressedIn().empty())
   {
     xyzElem->GetAttribute("expressed_in")->Set<std::string>(
-        this->XyzExpressedIn());
+        this->XyzExpressedIn(), _errors);
```
Comment (aaronchongth) [inline src/JointAxis.cc:402]: Ah coverage is complaining about this line. Perhaps testing with `JointAxis::SetXyzExpressedIn` would make it go away

**Review by aaronchongth — CHANGES_REQUESTED** [review]
Thanks! I've left a few test related comments, but otherwise looks good!

File: include/sdf/JointAxis.hh (hunk 1 above)
Comment (marcoag) [reply include/sdf/JointAxis.hh:239]: Added a test for `ToElement`.

File: src/JointAxis.cc [hunk 3]
```diff
@@ -374,37 +375,54 @@ sdf::ElementPtr JointAxis::Element() const
...
-  sdf::ElementPtr limitElem = axisElem->GetElement("limit");
-  limitElem->GetElement("lower")->Set<double>(this->Lower());
-  limitElem->GetElement("upper")->Set<double>(this->Upper());
-  limitElem->GetElement("effort")->Set<double>(this->Effort());
-  limitElem->GetElement("velocity")->Set<double>(this->MaxVelocity());
-  limitElem->GetElement("stiffness")->Set<double>(this->Stiffness());
-  limitElem->GetElement("dissipation")->Set<double>(this->Dissipation());
+  sdf::ElementPtr dynElem = axisElem->GetElement("dynamics", _errors);
+  dynElem->GetElement("damping", _errors)->Set<double>(
+      this->Damping());
```
Comment (aaronchongth) [inline src/JointAxis.cc:406]: Sorry I didn't notice this last review, I think you might have missed using the new overloaded function with errors here

File: src/JointAxis.cc [hunk 4]
```diff
@@ -374,37 +375,54 @@ sdf::ElementPtr JointAxis::Element() const
...
+  dynElem->GetElement("friction", _errors)->Set<double>(
+      this->Friction());
```
Comment (aaronchongth) [inline src/JointAxis.cc:408]: Same for this line as well

**Review by aaronchongth — CHANGES_REQUESTED** [review]
It should be just these 2 lines, thanks!

File: src/JointAxis.cc (hunk 3 above)
Comment (marcoag) [reply src/JointAxis.cc:406]: Good catch, thanks!

File: src/JointAxis.cc (hunk 4 above)
Comment (marcoag) [reply src/JointAxis.cc:408]: Good catch, thanks!

File: src/JointAxis.cc [hunk 5]
```diff
@@ -121,35 +122,35 @@ Errors JointAxis::Load(ElementPtr _sdf)
...
+    this->dataPtr->springReference = dynElement->Get<double>(errors,
+        "spring_reference", this->dataPtr->springReference).first;
+    this->dataPtr->springStiffness = dynElement->Get<double>(errors,
+        "spring_stiffness", this->dataPtr->springStiffness).first;
   }
 
   // Load limit values
   if (_sdf->HasElement("limit"))
   {
     sdf::ElementPtr limitElement = _sdf->GetElement("limit");
```
Comment (aaronchongth) [inline src/JointAxis.cc:140]: I believe this line might have missed out passing in `errors`

File: src/JointAxis.cc [hunk 6]
```diff
@@ -374,37 +375,54 @@ sdf::ElementPtr JointAxis::Element() const
   return this->dataPtr->sdf;
 }
 
-/////////////////////////////////////////////////
```
Comment (aaronchongth) [inline src/JointAxis.cc:377]: probably accidentally removed this line

**Review by aaronchongth — CHANGES_REQUESTED** [review]
final 2 small things, otherwise this looks great!

File: src/JointAxis.cc (hunk 6 above)
Comment (marcoag) [reply src/JointAxis.cc:377]: True, added back, thanks.

File: src/JointAxis.cc (hunk 5 above)
Comment (marcoag) [reply src/JointAxis.cc]: Yup, thanks for the catch, added.

**Review by aaronchongth — APPROVED** [review]
LGTM! Pending branch update and CI

**Pull request merged**